lib/streamlit/runtime/caching/hashing.py

```diff
@@ -451,6 +451,46 @@ def _to_bytes(self, obj: Any) -> bytes:
                 # it contains unhashable objects.
                 return b"%s" % pickle.dumps(obj, pickle.HIGHEST_PROTOCOL)
 
+        elif type_util.is_type(obj, "polars.series.series.Series"):
+            import polars as pl  # type: ignore[import-not-found]
+
+            obj = cast(pl.Series, obj)
+            self.update(h, obj.estimated_size())
+            self.update(h, obj.dtype.__hash__())
+
+            if len(obj) >= _PANDAS_ROWS_LARGE:
+                obj = obj.sample(n=_PANDAS_SAMPLE_SIZE, seed=0)
+
+            try:
+                self.update(h, obj.hash(seed=0).to_arrow().to_string().encode())
+                return h.digest()
+            except TypeError:
+                # Use pickle if polars cannot hash the object for example if
+                # it contains unhashable objects.
+                return b"%s" % pickle.dumps(obj, pickle.HIGHEST_PROTOCOL)
+        elif type_util.is_type(obj, "polars.dataframe.frame.DataFrame"):
+            import polars as pl
+
+            obj = cast(pl.DataFrame, obj)
+            self.update(h, obj.shape)
```

Comment:
nit: should we also include the `estimated_size` here similar to pl.Series? And maybe double check that it is not expensive and deterministic across reruns / sessions.

Reply:
I decided to include both `shape` and `type` both for series and data frames; I think they together eliminate the need to include esitmated_size, especially since [according to documentation](https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.estimated_size.html) the rules for computing it a little bit hairy, and probably could be not deterministic in edge cases.